src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1546 @@
+         * subjected to the same series of operations).
+         *
+         * @return a new object that is a copy of this generator
+         */
+        ArbitrarilyJumpableGenerator copy();
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a distance equal to 2<sup>{@code logDistance}</sup> within
+         * its state cycle.
+         *
+         * @param logDistance the base-2 logarithm of the distance to jump forward within the state
+         *                    cycle
+         *
+         * @throws IllegalArgumentException if {@code logDistance} is
+         *                                  2<sup>{@code logDistance}</sup> is
+         *                                  greater than the period of this generator
+         */
+        void jumpPowerOfTwo(int logDistance);
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a specified distance within its state cycle.
+         *
+         * @param distance the distance to jump forward within the state cycle
+         *
+         * @throws IllegalArgumentException if {@code distance} is not greater than
+         *                                  or equal to 0.0, or is greater than the
+         *                                  period of this generator
+         */
+        void jump(double distance);
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a large, fixed distance (typically 2<sup>64</sup> or more)
+         * within its state cycle. The distance used is that returned by method
+         * {@link ArbitrarilyJumpableGenerator#jumpDistance() jumpDistance}().
+         *
+         * @implSpec The default implementation invokes jump(jumpDistance()).
+         */
+        default void jump() { jump(jumpDistance()); }
+
+        /**
+         * Returns an effectively unlimited stream of new pseudorandom number
+         * generators, each of which implements the
+         * {@link ArbitrarilyJumpableGenerator} interface, produced by jumping
+         * copies of this generator by different integer multiples of the
+         * specified jump distance.
+         *
+         * @param distance a distance to jump forward within the state cycle
+         *
+         * @return a stream of objects that implement the {@link RandomGenerator} interface
+         *
+         * @throws IllegalArgumentException if {@code distance} is not greater than
+         *                                  or equal to 0.0, or is greater than the
+         *                                  period of this generator
+         *
+         * @implSpec The default implementation is equivalent to
+         * {@link ArbitrarilyJumpableGenerator#jumps(long) jumps}
+         * ({@link Long#MAX_VALUE Long.MAX_VALUE}).
+         */
+        default Stream<ArbitrarilyJumpableGenerator> jumps(double distance) {
```

Comment:
Suggest adding an Override annotation.

Reply:
jump(double distance) does not occur in superinterface. The ability to jump arbitrarily is specific to ArbitrarilyJumpableGenerator.